app/controllers/users_controller.rb

```diff
@@ -74,9 +79,11 @@ def update_membership(membership_params)
   end
 
   def parse_roster_ids(attrs)
-    attrs[:rosters] = attrs[:rosters].map do |roster_id|
-      Roster.find_by id: roster_id
-    end.compact
+    if attrs[:rosters].present?
+      attrs[:rosters] = attrs[:rosters].map do |roster_id|
+        Roster.find_by id: roster_id
+      end.compact
```

Comment:
`compact` returns a copy of self with all `nil` elements removed.

Reply:
Right, although that's after the `.map` block. An array with some `nil`s in it is still present.

However, because you're looking for rosters by id and no `Roster` will have a `nil` `id`, this can all just be done with where:

```ruby
def parse_roster_ids(attrs)
  attrs[:rosters] = Roster.where(id: attrs[:rosters]
  attrs
end
```

Reply:
+1 for Justin